backend/GymBeam/Controllers/UserController.cs

```diff
@@ -27,97 +28,175 @@ public UserController(IMediator mediator)
 #endif
         public async Task<ActionResult<IEnumerable<UserResponse>>> Get()
         {
-            var request = new GetAllUsersQuery();
-            var result = await _mediator.Send(request);
-            return Ok(result);
+            try
+            {
+                var request = new GetAllUsersQuery();
+                var result = await _mediator.Send(request);
+                return Ok(result);
+            }
+            catch (UserNotFoundException ex)
+            {
+                return StatusCode((int)HttpStatusCode.NotFound,
+                   $"NotFound: {ex.Message}");
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError,
+                    $"Internal Server Error: {ex.Message}");
+            }
         }
 
         [HttpGet("{id:int}")]
 #if !DEBUG
         [Authorize(Roles = Roles.Admin)]
 #endif
-        public ActionResult<UserResponse> Get(int id)
+        public async Task<ActionResult<UserResponse>> Get(int id)
         {
-            return new UserResponse
+            var request = new GetUserQuery
             {
-                Id = id,
-                Name = "testUsername",
-                DisplayName = "testDisplayName",
-                Role = "User"
+                UserId = id
             };
+            try
+            {
+                var result = await _mediator.Send(request);
+                return Ok(result);
+            }
+            catch (UserNotFoundException ex)
+            {
+                return StatusCode((int)HttpStatusCode.NotFound,
+                    $"NotFound: {ex.Message}");
+            }
+            catch (Exception ex)
+            {
+                return StatusCode((int)HttpStatusCode.InternalServerError,
+                    $"Internal Server Error: {ex.Message}");
+            }
         }
 
         [HttpGet("LoggedIn")]
 #if !DEBUG
         [Authorize(Roles = Roles.User)]
 #endif
-        public ActionResult<UserResponse> GetLoggedIn()
+        public async Task<ActionResult<UserResponse>> GetLoggedIn()
         {
-            int userId;
+            int id;
             try
             {
                 string cookiesUserId = Request.Cookies[Cookies.UserId];
```

Comment:
what if Cookies.UserId is not present in Cookies dictionary? It will throw exception.
Please do it in saver way
Cookies.TryGetValue()
fix that everywhere respectively